woocommerce/woocommerce-gutenberg-products-block
# Product Gallery block: Add support to Interactivity API

## What
The Product Gallery block serves as a container for several inner blocks that share a common context. As we progress towards enhancing the interactivity of these blocks, it's essential to integrate support for the Interactivity API within the Product Gallery block. 

Fixes #10715

## Why
This will enable all inner blocks to seamlessly share the block's context and state, thereby ensuring synchronized updates.



## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. From your WordPress dashboard, go to Appearance > Themes. Make sure you have a block-based theme installed and activated. If not, you can install one from the Add New option. Block-based themes include "Twenty-twenty Two," "Twenty-twenty Three," etc.
2. On the left-hand side menu, click on Appearance > Editor
3. On the left-hand side menu, click on Templates. 
4. Find and select the 'Single Product' template from the list.
5. When the Classic Product Template renders, click on Transform into Blocks. This will transform the Classic template in a block template if you haven't done it before.
6. Click on the plus icon (+) at the top left corner or within the editor to add a new block. In the search bar that appears, type Product Gallery and click on it to add the block to the template.
7. On the top-right side, click on the Save button.
8. Visit a product and inspect the Product Gallery block. Make sure you see the Interactivity API directives added to the block

![CleanShot 2023-08-23 at 21 34 14@2x](https://github.com/woocommerce/woocommerce-blocks/assets/20469356/46c84354-2f7b-429d-8289-01543640df74)




## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:
